Fix crash related to "delete all clusters", thermal & renewable

flomnes opened the pull request:
Fix #297 

* `clusterCount` becomes a method
* Use `std::vector` instead of C-style arrays.

flomnes committed:
Fix #297 for thermal clusters

flomnes committed:
Fix #297 for renewable clusters

flomnes committed:
Fix previous commits

flomnes committed:
Revert "Fix previous commits"

This reverts commit 2e3220906f2f87eedd78918b8de81a91cbdf069e.

guilpier-code commented on src/libs/antares/study/parts/renewable/container.h:
```diff
@@ -106,11 +106,13 @@ class PartRenewable
     // Only available from the solver
-    uint clusterCount;
-
+    inline uint clusterCount() const
```
I have a warning here : 
`container.h(131,1): warning C4267: 'return': conversion from 'size_t' to 'uint', possible loss of data`

flomnes replied:
OK

guilpier-code commented on src/libs/antares/study/parts/thermal/container.h:
```diff
@@ -124,11 +124,12 @@ class PartThermal
+    std::vector<ThermalCluster*> clusters;
 
+    // Return unit count
+    inline uint clusterCount() const {
```
`container.h(131,1): warning C4267: 'return': conversion from 'size_t' to 'uint', possible loss of data`

flomnes replied:
OK

guilpier-code commented on src/libs/antares/study/parts/renewable/container.cpp:
```diff
@@ -38,7 +38,7 @@ namespace Antares
 namespace Data
 {
-PartRenewable::PartRenewable() : list(), clusters(nullptr), clusterCount((uint)-1)
+PartRenewable::PartRenewable() : list(), clusters()
```
Do we really need the call to default `clusters()` constructor ?

flomnes replied:
Right, I also removed the call to other default ctors.

guilpier-code commented on src/libs/antares/study/parts/thermal/container.cpp:
```diff
@@ -39,12 +39,7 @@ namespace Antares
- mustrunList(),
- clusters(nullptr),
- clusterCount((uint)-1)
+ unsuppliedEnergyCost(0.), spilledEnergyCost(0.), list(), mustrunList(), clusters()
```
Same thing as above : 
Do we really need the call to default `clusters()` constructor ?

flomnes replied:
No, I removed this call and others.

flomnes committed:
PR remarks

- Do not call the default constructor
- clusterCount() returns a size_t instead of uint
